### ISSUE TITLE
Trouble connecting to RDS (Remote Desktop Services)

### ISSUE THREAD

**Issue by TSJasonH** [netbirdio/netbird#1653]
**Describe the problem**

Some users cannot connect to RDS, or have sporadic issues connecting. 
RDS uses a gateway connection over port 443/tcp and establishes connections to brokers on the backend for an RDP connection.

**To Reproduce**

1. Connect to Netbird
2. Click on RDP connection
3. Enter authentication
4. RDP Client immediately reports "There was a problem connecting to the remote resource. Ask your network administrator for help."
![RDPNBFail](https://github.com/netbirdio/netbird/assets/25849774/2b6c4de1-c6b7-4d23-9da5-3f4848611fe2)

**Expected behavior**

1. Connect to Netbird
2. Open RDP connection
3. Enter authentication
4. RDP client negotiates connection and opens desktop. 

**Are you using NetBird Cloud?**

No, self-hosted docker instance. 

**NetBird version**
0.26.2 across the board
(this issue was present in previous versions)

**NetBird status -d output:**

Everything is connected when checking status -d. 
The traceroute to the resource works as expected. It hops through the netbird peer we have setup to route this network exactly as it should. 

Even the RDWeb webpage at the same address as the RDS gateway will load just fine for the user. The issue happens when attempting to connect with the RDP client. 
The RDP connection file is identical and distributed to the users so there is no variance there.
All users have no problem accessing websites over the same NB connection. 

If the users connect to Wireguard the RDP connection works flawlessly every time. 
Does netbird do anything special with connections on port 443? 

The really weird thing is that it does work for some users (but sometimes not). 

I have tested this registry change https://github.com/netbirdio/netbird/issues/730 on some users laptops and it seems to resolve the problem some of the time. I'm uncertain if that is a red herring because it didn't fix it 100% of the time. 

I also tried changing some options in the /etc/sysconfig/netbird file on the router peer. Right now I've settled on these options:
NB_ICE_KEEP_ALIVE_INTERVAL_SEC=15
NB_ICE_DISCONNECTED_TIMEOUT_SEC=45
NB_ICE_FORCE_RELAY_CONN=false

Grasping at straws at this point.

**Comment by nazarewk** [comment]
@TSJasonH is this still an issue for you?

**Comment by TSJasonH** [comment]
Yes it's a pervasive issue for many of my users. They can't use NB and instead have to depend on a full WG tunnel. We're still running on 0.28.4.

**Comment by PizzaLovingNerd** [comment]
This was last reproduced on `v0.28.4`. Since then, [PR #4735](https://github.com/netbirdio/netbird/pull/4735) added MSS clamping for routed outbound TCP traffic to prevent path-MTU failures. This should address the reported RDS Gateway behavior.

Please test with a current client release and reopen the issue if the problem persists. MTU configuration is available through the [`--mtu` client option](https://docs.netbird.io/get-started/cli), and related advanced settings are documented under [client environment variables](https://docs.netbird.io/client/environment-variables#firewall).

Cheers!
